src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -5673,13 +5913,19 @@ init_myhook(void)
 	else
 		daos_debug_inited = true;
 
-	d_agetenv_str(&env_log, "D_IL_REPORT");
+	rc = d_agetenv_str(&env_log, "D_IL_REPORT");
 	if (env_log) {
 		report = true;
 		if (strncmp(env_log, "0", 2) == 0 || strncasecmp(env_log, "false", 6) == 0)
 			report = false;
 		d_freeenv_str(&env_log);
+	} else if (rc == -DER_NONEXIST) {
+		d_setenv("D_IL_REPORT", "0", 1);
```

Comment:
I don't think you should be setting these.  I assume the intention is to save them for if required later in exec but that has two problems:

1. They might have been stripped from the environment
2. When running pil4dfs then the child and grandchild will have different environment

Assuming the intention here is to save them for later then use a static global variable for this.

Reply:
same as the above comment. 

It would lead to better consistency to let the lib/application set default value if an env is not set instead of using a saved value at the initialization phase.

Assuming env_var's default value is 0.
bash runs a bash script with env_var=1.
"unset env_var" is invoked inside bash script to clear env on purpose.
"env_var=1" will still be used in following new processes.

For the case of creating a child process with stripped env, parent process does not unset env in environ. Instead, the parent process just passes a new env list in exec().

Reply:
The above discussion was in the context of a PR where execv was being modified but now the PR only modified execve.

I still don't see any case for setting this here.  If a user has set it and it's propagated then fine, but why would a process turn it on for it's child processes?